polkadot/node/core/pvf/prepare-worker/src/lib.rs

```diff
@@ -229,31 +234,74 @@ pub fn worker_entrypoint(
 					},
 				};
 
-				// SAFETY: new process is spawned within a single threaded process. This invariant
-				// is enforced by tests.
-				let result = match unsafe { nix::unistd::fork() } {
-					Err(errno) => Err(error_from_errno("fork", errno)),
-					Ok(ForkResult::Child) => {
-						// Dropping the stream closes the underlying socket. We want to make sure
-						// that the sandboxed child can't get any kind of information from the
-						// outside world. The only IPC it should be able to do is sending its
-						// response over the pipe.
-						drop(stream);
-						// Drop the read end so we don't have too many FDs open.
-						drop(pipe_reader);
-
-						handle_child_process(
-							pvf,
-							pipe_writer,
-							preparation_timeout,
-							prepare_job_kind,
-							executor_params,
-						)
-					},
-					Ok(ForkResult::Parent { child }) => {
-						// the read end will wait until all write ends have been closed,
-						// this drop is necessary to avoid deadlock
-						drop(pipe_writer);
+				cfg_if::cfg_if! {
+					if #[cfg(any(target_os = "linux"))] {
+						let stack = 1024 * 1024;
```

Comment:
This is not enough, it should be at least the native stack limit (see `DEFAULT_CONFIG`). I think we can use `EXECUTE_THREAD_STACK_SIZE` here instead.

And now that I see it, `EXECUTE_THREAD_STACK_SIZE` doesn't correctly get updated based on the executor parameters. It should depend on the output of `params_to_wasmtime_semantics`, in other words it shouldn't be a constant anymore. The execute worker should already have access to the executor parameters. Can you make that change here? Hope it makes sense. 😅

Reply:
I'm going to change this.